pkg/dns/dns.go

```diff
@@ -48,28 +49,38 @@ const (
 
 	// Resync period for the kube controller loop.
 	resyncPeriod = 30 * time.Minute
+
+	// Duration for which the TTL cache should hold the node.
+	nodeCacheTTL = 180 * time.Second
```

Comment:
Added a clarification comment. We are not setting a watch here. We just need one node and our assumption is that the zone name annotation on the nodes don't change often, so there is not really a need to observe and react to the nodes' state changes. But we don't want to hold the values for ever without letting it ever change. So the value is chosen to be neither too long nor too short.